### ISSUE TITLE
Publish documentation

### ISSUE THREAD

**Issue by renehernandez** [roboll/helmfile#1755]
I want to propose publishing versioned docs of helmfiles. A solution such as:
* Github Pages released from a 
* mkdocs
* versioned by released tag (with `latest` pointing to the most recent tag)
* deploys automatically on every tag release

would be a great starting place.

Would there be any other requirements out of the box? Otherwise, I can start working on a PR for the implementation

**Comment by mumoshu** [comment]
@renehernandez Hey! Thanks for suggestion. But I'm still on hold for this. I believe we don't even have enough love and content in our documentation yet. Let's add more content and publish it in a versioned way only after we are confident that we do have enough love and human resource to maintain it in a meaningful way.

**Comment by renehernandez** [comment]
One thing that I forgot to mention and I think is the main benefit of docs, is that it makes it easy to search, while right now, I have to look at the readme and each file individually on the docs folder.

And the documentation size shouldn't matter that much to decide whether we should automate and publish helmfile docs. IMO, I believe that the sooner we start the better.

Example: I am building a small utility [appfile](https://github.com/renehernandez/appfile) and I have alreadye published the docs at https://github.com/renehernandez/appfile

**Comment by renehernandez** [comment]
@mumoshu This is an example of how the documentation would look like as a first step: https://renehernandez.github.io/helmfile

I know that you don't want to dedicate resources to this for now, but if still makes sense I can open the initial PR that would deploy this automatically

**Comment by mumoshu** [comment]
@renehernandez Maybe it's worth trying then...?

I have no experience with `mkdocs`. In `appfile` and its `docs` directory you seem to be using some Golang script to run `mkdocs`, right?

Is there any mature alternative to `mkdocs` that is written in Go? Or should I just go with `mkdocs` believing it's mature enough and will never break?

**Comment by renehernandez** [comment]
@mumoshu Yes, `mkdocs` is a well used and maintained documentation system. I have experience with the system both in open source and internally at work.

I am OK with taking a leading role on this part of the system.

I'll get you a PR by tomorrow

**Comment by renehernandez** [comment]
> Is there any mature alternative to mkdocs that is written in Go?

Of that I am not aware, unfortunately

**Comment by renehernandez** [comment]
> I have no experience with mkdocs. In appfile and its docs directory you seem to be using some Golang script to run mkdocs, right?

I missed this one. I am using a golang script to generate the CLI reference for `appfile` using some functionalities provided by `cobra`. I looked around and in the code, and I would have to do some changes before providing something similar for `helmfile`. That will come in a later PR
